lib/ts/recipe/emailpassword/api/implementation.ts

```diff
@@ -193,14 +234,37 @@ export default function getAPIImplementation(): APIInterface {
                 };
             }
 
+            if (linkingCandidate.isPrimaryUser && emailPasswordAccount !== undefined) {
```

Comment:
Add test: SSO with email A, and ep with email A (both are linked). When generating the password reset, it should not call should do account linking, and generate a link. WHen consumingt he token, it should also not call should do accoun tlinking, and it should change the password